typescript/sdk/src/token/EvmERC20WarpRouteReader.ts

```diff
@@ -1,33 +1,58 @@
-import { ethers, providers } from 'ethers';
+import { providers } from 'ethers';
 
 import {
   ERC20__factory,
+  HypERC20CollateralVaultDeposit__factory,
   HypERC20Collateral__factory,
+  HypERC20__factory,
 } from '@hyperlane-xyz/core';
-import { ERC20Metadata, ERC20RouterConfig } from '@hyperlane-xyz/sdk';
-import { Address } from '@hyperlane-xyz/utils';
+import {
+  ERC20Metadata,
+  TokenRouterConfig,
+  TokenType,
+} from '@hyperlane-xyz/sdk';
+import { Address, isZeroishAddress, rootLogger } from '@hyperlane-xyz/utils';
 
 import { DEFAULT_CONTRACT_READ_CONCURRENCY } from '../consts/concurrency.js';
 import { EvmHookReader } from '../hook/EvmHookReader.js';
-import { EvmIsmReader } from '../ism/EvmIsmReader.js';
+import {
+  DerivedIsmConfigWithAddress,
+  EvmIsmReader,
+} from '../ism/EvmIsmReader.js';
 import { MultiProvider } from '../providers/MultiProvider.js';
-import { ChainName } from '../types.js';
+import { ChainNameOrId } from '../types.js';
 
 type WarpRouteBaseMetadata = Record<
-  'mailbox' | 'owner' | 'token' | 'hook' | 'interchainSecurityModule',
+  'mailbox' | 'owner' | 'token' | 'hook',
   string
+> & { interchainSecurityModule?: DerivedIsmConfigWithAddress };
+
+/**
+ * @remark
+ * We only expect to support deriving a subset of these types, for now.
```

Comment:
`DerivedTokenType` applies to this Module because we can only support deriving these 4 TokenTypes through **specific** on-chain method calls. I believe we discussed this a a while back